internal/controller/persistentvolumeclaim_controller.go

```diff
@@ -66,16 +57,57 @@ func (r *PersistentVolumeClaimReconciler) Reconcile(ctx context.Context, req ctr
 		}, nil
 	}
 
-	// Skip if the PVC is not deleted or PVC does not have TopoLVM's finalizer.
-	if pvc.DeletionTimestamp == nil {
+	// Skip if PVC does not have TopoLVM's finalizer.
+	if !controllerutil.ContainsFinalizer(pvc, topolvm.PVCFinalizer) {
 		return ctrl.Result{}, nil
 	}
 
-	if !controllerutil.ContainsFinalizer(pvc, topolvm.PVCFinalizer) {
-		return ctrl.Result{}, nil
+	if pvc.DeletionTimestamp == nil {
+		return ctrl.Result{}, r.onUpdate(ctx, pvc)
+	}
+
+	return r.onDelete(ctx, pvc)
+}
+
+func (r *PersistentVolumeClaimReconciler) onUpdate(ctx context.Context, pvc *corev1.PersistentVolumeClaim) error {
+	log := crlog.FromContext(ctx)
+	// If the PVC has `FileSystemResizePending` condition,
+	// annotate relating pods to notify kubelet to resize the filesystem.
+	// Note: kubelet periodically checks pods to resize fs but it may take a long time.
+	// Manually annotating pods is a way to speed up the process.
+	for _, cond := range pvc.Status.Conditions {
+		if cond.Type == corev1.PersistentVolumeClaimFileSystemResizePending && cond.Status == corev1.ConditionTrue {
+			pods, err := r.getPodsByPVC(ctx, pvc)
+			if err != nil {
+				log.Error(err, "unable to fetch PodList for a PVC", "pvc", pvc.Name, "namespace", pvc.Namespace)
+				return err
+			}
+			for _, pod := range pods {
+				pod.Annotations["topolvm.io/resizefs-notify"] = time.Now().String()
```

Comment:
I think we should set this annotation with a docs entry in addition to the time to make it clear why we do this. apart from that I think its a good idea and we should keep it

Reply:
Do you mean adding a docs annotation to pods?

Reply:
I was thinking of changing the current annotation because its not used anywhere else and serves only as marker. Also I think we should add it as part of our normal docs

Reply:
IC. This is a draft, I'm going to work on it more. When I'm done, I'll assign you as a reviewer.